web/web.go

```diff
@@ -0,0 +1,771 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Package web defines web APIs to be used with harbourbridge frontend.
+// Apart from schema conversion, this package involves API to update
+// converted schema.
+
+package web
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	//"harbourbridge-web/models"
+	"github.com/cloudspannerecosystem/harbourbridge/conversion"
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/mysql"
+	"github.com/cloudspannerecosystem/harbourbridge/postgres"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/gorilla/handlers"
+	_ "github.com/lib/pq"
+)
+
+// TODO(searce):
+// 1) Test cases for APIs
+// 2) API for saving/updating table-level changes.
+// 3) API for showing logs
+// 4) Split all routing to an route.go file
+// 5) API for downloading the schema file, ddl file and summary report file.
+// 6) Update schema conv after setting global datatypes and return conv. (setTypeMap)
+// 7) Add rateConversion() in schema conversion, ddl and report APIs.
+// 8) Add an overview in summary report API
+var mysqlTypeMap = make(map[string][]typeIssue)
```

Comment:
In general, files in go should be organized with the important stuff at the top. The goal is to make the file readable and easy to browse. That usually means:
- public constants first
- key public type definitions next (although often it makes sense to put them next to public functions that use them)
- then public functions (and relevant type definitions)
- and helper functions and other non-public definitions last (generally in order of importance)

Reply:
Yes, agreed. Let's add:

// TODO: organize this file according to go style guidelines: generally
// have public constants and public type definitions first, then public
// functions, and finally helper functions (usually in order of importance).

and we can resolve this thread.

Reply:
added accordingly